src/normalization.jl

```diff
@@ -106,11 +106,35 @@ function imsharpen(
 
     image_equalized_gray = Gray.(image_equalized)
 
+    return unsharp_mask(image_equalized_gray, smoothing_param, intensity)
+end
+
+"""
+    unsharp_mask(image_equalized_gray, smoothing_param, intensity)
+
+Apply unsharp masking on (equalized) grayscale ([0, `clapmax`]) image to enhance its sharpness.
+
+# Arguments
+- `image_equalized_gray: The input grayscale image, typically already equalized.
+- `smoothing_param::Int`: The pixel radius for Gaussian blurring (typically between 1 and 10).
+- `intensity: The amount of sharpening to apply. Higher values result in more pronounced sharpening.
+
+# Returns
+The sharpened grayscale image with values clipped between 0 and `clapmax`.
+"""
+function unsharp_mask(image_equalized_gray, smoothing_param, intensity, clampmax)
     image_smoothed = imfilter(image_equalized_gray, Kernel.gaussian(smoothing_param))
+    clamp!(image_smoothed, 0.0, clampmax)
```

Comment:
The goal is to make sure that the input image is still within the range of allowed values, right? Allowing for the image data to be either 0-1 or from 0-255. My understanding is that gaussian smoothing cannot increase the maximum value of an array or decrease the minimum value. In what case is this first call to `clamp` needed?

Reply:
I ran the algorithm on test images and was getting negative values, so clipping is necessary at least for the lower bound.

Reply:
interesting, makes sense